openedx/core/djangoapps/embargo/views.py

```diff
@@ -2,10 +2,36 @@
 
 from django.http import Http404
 from django.views.generic.base import View
+from edx_rest_framework_extensions.authentication import JwtAuthentication
+from rest_framework import permissions
+from rest_framework.views import APIView
 
 from edxmako.shortcuts import render_to_response
 
 from . import messages
+from .api import check_course_access
+from .permissions import CanCallCheckCourseAccessAPI
+
+
+class CheckCourseAccessView(APIView):
+    authentication_classes = (JwtAuthentication, )
+    permission_classes = (permissions.IsAuthenticated, CanCallCheckCourseAccessAPI)
+
+    def post(self, request):
```

Comment:
Maximum length I've seen is 2048 characters. Usernames have a maximum length of 30 characters. Our current use case will have us calling this for a maximum of 10 course runs. Even if our course keys were 100 characters long, we would still have plenty of leeway. Besides this our API guidelines (search on wiki, I'm on mobile) state that POST is used to create. 

Privacy is a concern but POST is not the solution here. If we don't trust our engineers, who already have access to the production database, we have a bigger problem than this API endpoint. 

Please use GET.